packages/batch-poster-monitor/index.ts

```diff
@@ -267,6 +275,47 @@ const checkForUserTransactionBlocks = async ({
   return userTransactionBlockFound
 }
 
+const getBatchPostingTimeBounds = async (
+  childChainInformation: ChainInfo,
+  parentChainClient: PublicClient
+) => {
+  let batchPostingTimeBounds = BATCH_POSTING_TIMEBOUNDS_FALLBACK
+  try {
+    const maxTimeVariation = await parentChainClient.readContract({
+      address: childChainInformation.ethBridge.sequencerInbox as `0x${string}`,
+      abi: parseAbi([
+        'function maxTimeVariation() view returns (uint256, uint256, uint256, uint256)',
+      ]),
+      functionName: 'maxTimeVariation',
+    })
+
+    const delayBlocks = Number(maxTimeVariation[0])
+    const delaySeconds = Number(maxTimeVariation[2].toString())
+
+    // use the minimum of delayBlocks or delay seconds
+    batchPostingTimeBounds = Math.min(
+      delayBlocks *
+        getParentChainBlockTimeForBatchPosting(childChainInformation),
+      delaySeconds
+    )
+
+    console.log({
+      delayBlocks,
+      delaySeconds,
+      batchPostingTimeBounds,
+    })
+  } catch (_) {
+    // no-op, use the fallback value
+  }
+
+  // formula : min(50% of x , max(1h, x - 18h))
```

Comment:
The constant seems to be 9h instead of 18h

Reply:
Yes I was adjusting the buffer values to get a reasonable timebound. 9 seems to be working better than 18, but we might need to change it to 6 as well. Will discuss it with the broader team.